examples/precondition.py

```diff
@@ -0,0 +1,39 @@
+r"""This example will show you how to use preconditions."""
+
+import glob
+import os
+import numpy as np
+
+from dawgz import job, after, waitfor, require, ensure, schedule
+
+
+@ensure(lambda i: os.path.exists(f'normal-{i}.npy'))
+@job(array=10)
+def generate_normal(i: int):
+    np.save(f'normal-{i}.npy', np.random.normal())
+
+@ensure(lambda i: os.path.exists(f'uniform-{i}.npy'))
+@job(array=100)
+def generate_uniform(i: int):
+    np.save(f'uniform-{i}.npy', np.random.random())
+
+@job
+def print_intro():
+    print('Hello world!')
+
+@after(generate_normal)
+@after(generate_uniform)
+@after(print_intro)
+@require(lambda: len(glob.glob('*.npy')) >= 10 + 100)  # Precondition that depends on several dependencis
```

Comment:
This will always be true in this example because you wait for the success of `generate_normal` an `generate_uniform`. Using `status='any'` for these two with `len(...) > 69` could be more instructive. Also `print_intro` is useless.

Reply:
Something like this ? I am not sure it works actually because I don't know if the status of an array works like I think it does (returns `FAILED` if a single index has failed).

```python
#!usr/bin/env python

import glob
import os
import numpy as np

from dawgz import job, after, require, ensure, schedule

@ensure(lambda i: os.path.exists(f'samples_{i}.npy'))
@job(array=100)
def generate(i: int):
    np.save(f'samples_{i}.npy', np.random.random())

@after(generate, status='any')
@require(lambda: len(glob.glob('*.npy')) > 69)
@job
def merge():
    files = glob.glob('*.npy')
    stack = np.vstack([np.load(f) for f in files])

    np.save(f'samples.npy', stack)

schedule(merge, backend='local')
```